RPC.en.rst

```diff
@@ -3,59 +3,257 @@
 .. highlight:: cpp
 .. default-domain:: cpp
 
-.. _rpc_refactor:
+.. _rpc:
 
-RPC Refactor
+RPC
 *************
 
+Overview
+========
+
 To communicate between ``traffic_manager`` and ``traffic_server`` there is an RPC mechanism in
-``mgmt/api``. This is a simple serialization style RPC which runs over sockets. Current the logic is
-tightly embedded in the management library. It would be useful for various reasons to extract this
-in to a separate library.
+``mgmt/``. This is a simple serialization style RPC which runs over sockets.
+
+
+Serialization Mechanism
+=======================
+
+A RPC message is sent as a ``MgmtMessageHdr`` followed by the serialized data in bytes. 
+
+.. class:: MgmtMarshall
+
+   This is the class used to marshall data objects. It provides functions to marshall and unmarshall data. Each data object is associated with a field. Fields can be:
+
+.. enum:: MGMT_MARSHALL_INT
+.. enum:: MGMT_MARSHALL_LONG
+.. enum:: MGMT_MARSHALL_STRING
+.. enum:: MGMT_MARSHALL_DATA
+
+
+Marshalling:
+++++++++++++
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_marshall(void *buf, size_t remain, const MgmtMarshallType *fields, unsigned count, ...)
+
+   Variable argument wrapper for ``mgmt_message_marshall_v``. Allows for different datatypes to be marshalled together as long as a field is specified for each data object.
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_marshall_v(void *buf, size_t remain, const MgmtMarshallType *fields, unsigned count, va_list ap)
 
-In addition there seems to be a delay between a command line tool such as ``traffic_ctl`` sending
-data and ``traffic_server`` receiving the data. This is likely due to :code:`Continuation`
-scheduling that causes the socket to be polled every few seconds.
+   This function goes through all the data objects and serializes them together into a buffer. Based on the field, the number of bytes is determined and if there is enough space, it is written into the buffer. 
 
-The phases of the project
+    - MGMT_MARSHALL_INT: 4 bytes.
```

Comment:
Why are these `-` instead of `*` as below? Also, because we use a 3 space indent in general use `* ` as the prefix for lists.

Reply:
I think i was trying something out and forgot to change it back.